index.js

```diff
@@ -48,11 +50,47 @@ function spawnClangFormat(args, done, stdio) {
               "Consider installing it with your native package manager instead.\n";
     throw new Error(message);
   }
-  var clangFormatProcess = spawn(nativeBinary, args, {stdio: stdio});
-  clangFormatProcess.on('close', function(exit) {
-    if (exit) done(exit);
-  });
-  return clangFormatProcess;
+
+  // extract glob, if present
+  var filesGlob = args.filter(function(arg){return arg.indexOf('--glob=') !== -1;})
+                    .map(function(arg){return arg.replace('--glob=', '');})
+                    .shift();
+
+  if (filesGlob) {
+    // remove glob from arg list
+    args = args.filter(function(arg){return arg.indexOf('--glob=') === -1;});
+
+    return glob(filesGlob, function(er, files) {
+      // launch a new process for each file, up to 200 at a time
+      async.parallelLimit(
+        files.map(function(file) {
+          return function(callback) {
+            var clangFormatProcess = spawn(nativeBinary,
```

Comment:
To be honest, I didn't check this at the start and when I did I assumed it would be a bad idea to run the binary with a couple hundred individual files.

That might be a bad assumption though. I'll try running `clang-format --glob=node_modules/**/*.js` and see if it crashes horribly.